Add a Grunt action to set GP_SCRIPT_DEBUG to false when we deploy to wporg

amieiro opened the pull request:
<!-- 
Thanks for contributing to GlotPress! Please, follow the GlotPress Contributing Guidelines:
https://github.com/GlotPress/GlotPress/blob/develop/CONTRIBUTING.md 
-->

## Problem

When we deploy a new version to wporg, we need to minify the CSS and the JavaScript tiles. 

- In https://github.com/GlotPress/GlotPress/pull/1614 we added a `GP_SCRIPT_DEBUG` constant to load the unminified CSS and JavaScript files by default, setting this `GP_SCRIPT_DEBUG` constant to `true` in the **glotpress.php** file.
- In https://github.com/GlotPress/GlotPress/pull/1654 and in https://github.com/GlotPress/GlotPress/pull/1655 we added some actions to minify this files when we deploy to wporg.

But we need to change the `GP_SCRIPT_DEBUG` constant to `false` before we deploy to wporg, so the wporg release will use the minified files (except the end user will use the `SCRIPT_DEBUG` set to `true`). This PR address this problem.

Fixes https://github.com/GlotPress/GlotPress/issues/1587.

<!--
Please, describe what is the status/problem before the PR.
-->

## Solution

This PR adds:
- A **Grunt action** to change the `define( 'GP_SCRIPT_DEBUG', true );` to `define( 'GP_SCRIPT_DEBUG', false );` in the glotpress.php file. **Gruntfile.js** file.
- A **npm action** to run the Grunt action. **package.json** file.
- This npm action into the **wporg workflow**. **.github/workflows/deploy-wporg-plugin.yml** file.

<!--
Please, describe how this PR improves the situation.
-->

## Testing Instructions

To test this PR:
- You can execute the Grunt action with the `grunt replace:updateGPScriptDebugConst` command.
- You can execute the npm action with the `npm run updateGPScriptDebugConst` command.

In both situations, you can check that the `GP_SCRIPT_DEBUG` value in the `glotpress.php` file has changed from true to false.
<!-- 
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
Please, include step-by-step instructions on how to test this PR:
1. Open a original string in a project.
2. Add the translation.
3. etc. 
-->

amieiro committed:
Add a Grunt action no set GP_SCRIPT_DEBUG to false when we deploy to wporg

amieiro committed:
Change the updateGPScriptDebug name to updateGPScriptDebugConst

amieiro committed:
Merge branch 'develop' into gp-script-debug-false

amieiro committed:
Merge branch 'develop' into gp-script-debug-false

amieiro committed:
Merge branch 'develop' into gp-script-debug-false

akirk commented on .github/workflows/deploy-wporg-plugin.yml:
```diff
@@ -28,6 +28,7 @@ jobs:
         run: |
           npm ci
           npm run build
+          npm run updateGPScriptDebugConst
```
```suggestion
          npm run prepare-release
```

akirk commented on Gruntfile.js:
```diff
@@ -67,6 +67,18 @@ module.exports = function( grunt ) {
 			},
 		},
+		replace: {
+			updateGPScriptDebugConst: {
```
```suggestion
			prepare-release: {
```

akirk commented on package.json:
```diff
@@ -19,11 +19,13 @@
   "scripts": {
     "build": "grunt",
     "watch": "grunt watch",
+    "updateGPScriptDebugConst": "grunt replace:updateGPScriptDebugConst",
```
```suggestion
    "prepare-release": "grunt replace:prepare-release",
```

akirk reviewed (commented):
What about this?

amieiro committed:
Update .github/workflows/deploy-wporg-plugin.yml

Co-authored-by: Alex Kirk <akirk@users.noreply.github.com>

amieiro committed:
Update Gruntfile.js

Co-authored-by: Alex Kirk <akirk@users.noreply.github.com>

amieiro committed:
Update package.json

Co-authored-by: Alex Kirk <akirk@users.noreply.github.com>

amieiro committed:
Merge branch 'develop' into gp-script-debug-false

amieiro committed:
Add single quotation marks to the Grunt action

The dash introduced with the new name causes problems executing the
Grunt task, so we need to add these single quotation marks

amieiro commented:
> What about this?

With the dash in the Grunt action name, I have problems running the task in my local machine, so I have added single quotation marks to this action name.

```
npm run prepare-release

> glotpress@0.0.0 prepare-release
> grunt replace:prepare-release

Loading "Gruntfile.js" tasks...ERROR
>> SyntaxError: Unexpected token '-'
Warning: Task "replace:prepare-release" not found. Use --force to continue.

Aborted due to warnings.
```

akirk reviewed (approved):
Thank you!
